DataStructures.Tests/BinaryTreeTests.cs

```diff
@@ -0,0 +1,127 @@
+﻿using DataStructures.Trees;
+using System.Text;
+using Xunit;
+
+namespace DataStructures.Tests
+{
+    public class TreeTests
+    {
+        [Fact]
+        public void Add_value_to_empty_tree()
+        {
+            // Arrange
+            BinarySearchTree tree = new BinarySearchTree();
+
+            // Act
+            tree.Add(4);
+
+            //Assert
+            Assert.Equal(4, tree.Root.Value);
+        }
+
+        [Fact]
+        public void Add_one_Left_and_one_Right_node()
+        {
+            // Arrange
+            BinarySearchTree tree = new BinarySearchTree();
+
+            // Act
+            tree.Add(8);
+            tree.Add(4);
+            tree.Add(5);
+
+            //Assert
+            Assert.Equal(8, tree.Root.Value);
+            Assert.Equal(4, tree.Root.Left.Value);
+            Assert.Equal(5, tree.Root.Right.Value);
+        }
+
+        [Fact]
+        public void Contains_a_value_to_find_Node()
+        {
+            // Arrange
+            BinarySearchTree tree = new BinarySearchTree();
+            tree.Add(8);
+            tree.Add(4);
+            tree.Add(5);
+            tree.Add(1);
+            tree.Add(6);
+            tree.Add(3);
+
+            // Act
+            bool trueResult = tree.Contains(tree.Root, 8);
+            bool falseResult = tree.Contains(tree.Root, 15);
+
+            //Assert
+            Assert.True(trueResult);
+            Assert.False(falseResult);
+        }
+
+        [Fact]
+        public void PreOrder_returns_strings()
+        {
+            // Arrange
+            BinarySearchTree tree = new BinarySearchTree();
+            tree.Add(8);
+            tree.Add(4);
+            tree.Add(5);
+            tree.Add(1);
+            tree.Add(6);
+            tree.Add(3);
+
+            StringBuilder sb = new StringBuilder();
+
+            // Act
+            string result = tree.PreOrder(sb, tree.Root);
+
+            //Assert
+            Assert.Equal("8 4 5 1 6 3", result);
```

Comment:
This is what your tree looks like:
........................8
..................4
.........1.............5
............3..............6
So your _actual_ result from the test is correct (as in your preorder is working correctly).  The proper string to go in the expected argument of your assert would be "8 4 1 3 5 6".